qiskit_nature/properties/electronic_structure/dipole_moment.py

```diff
@@ -0,0 +1,160 @@
+
+    @classmethod
+    def from_driver_result(cls, result: DriverResult) -> None:
+        """This property does not support construction from a driver result (yet).
+
+        Args:
+            result: ignored.
+
+        Raises:
+            NotImplemented
+        """
+        raise NotImplementedError()
+
+    def second_q_ops(self) -> List[FermionicOp]:
+        """Returns a list containing the dipole moment operator."""
+        return self._dipole.second_q_ops()
+
+    def interpret(self, result: EigenstateResult) -> None:
+        """Interprets an `qiskit_nature.result.EigenstateResult` in the context of this Property.
+
+        This is currently a method stub which may be used in the future.
+
+        Args:
+            result: the result to add meaning to.
+        """
+        raise NotImplementedError()
+
+
+class TotalDipoleMoment(Property):
+    """The TotalDipoleMoment property."""
+
+    def __init__(
+        self,
+        dipole_axes: Dict[str, DipoleMoment],
+        dipole_shift: Optional[Dict[str, DipoleTuple]] = None,
+    ):
+        """
+        Args:
+            dipole_axes: a dictionary mapping Cartesian axes to DipoleMoment properties.
+            dipole_shift: an optional dictionary of named dipole shifts.
+        """
+        super().__init__(self.__class__.__name__)
+        self._dipole_axes = dipole_axes
+        self._dipole_shift = dipole_shift
+
+    @classmethod
+    def from_driver_result(cls, result: DriverResult) -> Optional["TotalDipoleMoment"]:
+        """Construct a TotalDipoleMoment instance from a QMolecule.
+
+        Args:
+            result: the driver result from which to extract the raw data. For this property, a
+                QMolecule is required!
+
+        Returns:
+            An instance of this property.
+
+        Raises:
+            QiskitNatureError: if a WatsonHamiltonian is provided.
+        """
+        cls._validate_input_type(result, ElectronicDriverResult)
+
+        qmol = cast(QMolecule, result)
+
+        if not qmol.has_dipole_integrals():
+            return None
+
+        def dipole_along_axis(axis, mo_ints, energy_shift):
+            return DipoleMoment(
+                axis,
+                ElectronicEnergy(
+                    ElectronicBasis.MO,
+                    {1: OneBodyElectronicIntegrals(ElectronicBasis.MO, mo_ints)},
+                    energy_shift=energy_shift,
+                ),
+            )
+
+        return cls(
+            {
+                "x": dipole_along_axis(
+                    "x", (qmol.x_dip_mo_ints, qmol.x_dip_mo_ints_b), qmol.x_dip_energy_shift
+                ),
+                "y": dipole_along_axis(
+                    "y", (qmol.y_dip_mo_ints, qmol.y_dip_mo_ints_b), qmol.y_dip_energy_shift
+                ),
+                "z": dipole_along_axis(
+                    "z", (qmol.z_dip_mo_ints, qmol.z_dip_mo_ints_b), qmol.z_dip_energy_shift
+                ),
+            },
+            dipole_shift={
```

Comment:
I guess I was not expecting nuclear part in here as a 'shift'. If this is total dipole moment then given we compute the electronic part it needs this to make the total. 
So now I am going to comment on the name ElectronicEnergy and not it has similar with nuclear as a shift. But why then is this called ElectronicEnergy and not say just Energy - i.e although we only compute the electronic part the dipole that does not appear in its name.

Reply:
The `nuclear` shifts were added in view of the future change to include the `interpret` methods. This field is not being used now and we can discuss how to handle this when dealing with #243